layers/+tools/tide/funcs.el

```diff
@@ -0,0 +1,99 @@
+;;; funcs.el --- Tide  Layer functions File for Spacemacs
+;;
+;; Copyright (c) 2012-2020 Sylvain Benner & Contributors
+;;
+;; Author: Thanh Vuong <thanhvg@gmail.com>
+;; URL: https://github.com/thanhvg
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+
+(defun spacemacs//tide-setup-prefix ()
+  "This one should run only once, otherwise `which-key' will become very slow #12455"
+  (dolist (mode tide-managed-modes)
+    (spacemacs/declare-prefix-for-mode mode "m=" "format")
+    (spacemacs/declare-prefix-for-mode mode "mE" "errors")
+    (spacemacs/declare-prefix-for-mode mode "mg" "goto")
+    (spacemacs/declare-prefix-for-mode mode "mh" "help")
+    (spacemacs/declare-prefix-for-mode mode "mn" "name")
+    (spacemacs/declare-prefix-for-mode mode "mr" "refactor")
+    (spacemacs/declare-prefix-for-mode mode "mS" "server")
+    (spacemacs/declare-prefix-for-mode mode "ms" "send")))
+
+(defun spacemacs//tide-setup-bindings ()
+  "Define keys bindings for `tide-mode'"
+  (spacemacs/set-leader-keys-for-minor-mode 'tide-mode
+    "Ee" #'tide-fix
+    "Ed" #'tide-add-tslint-disable-next-line
+    "gb" #'tide-jump-back
+    "gg" #'tide-jump-to-definition
+    "gt" #'spacemacs/typescript-jump-to-type-def
+    "gr" #'tide-references
+    "hh" #'tide-documentation-at-point
+    "ri" #'tide-organize-imports
+    "rr" #'tide-rename-symbol
+    "rf" #'tide-rename-file
+    "Sr" #'tide-restart-server
+    "Sj" #'spacemacs//tide-create-jsconfig-file))
+
+(defun spacemacs//tide-setup ()
+  "Setup tide backend.
+Must be called by a layer using tide."
+  (progn
```

Comment:
This progn is not necessary, please remove.

Reply:
We want to keep the progns in the use-package declaration, however here it is really not necessary and should be removed.